provisioner/node_pools.go

```diff
@@ -64,6 +69,38 @@ type AWSNodePoolProvisioner struct {
 	logger          *log.Entry
 }
 
+type amiID struct {
+	ec2Client ec2iface.EC2API
+}
+
+func (a amiID) function() func(...string) (string, error) {
+	return func(params ...string) (string, error) {
+		if len(params) != 2 {
+			return "", fmt.Errorf("the template func %s take exactly 2 arguments: AMI name and owner ID", a.name())
+		}
+		filterNameName := describeImageFilterNameName
+		filterOwnerName := describeImageFilterNameOwner
+		name := params[0]
+		ownerAccount := params[1]
+		input := ec2.DescribeImagesInput{Filters: []*ec2.Filter{
+			{Name: &filterNameName, Values: []*string{&name}},
+			{Name: &filterOwnerName, Values: []*string{&ownerAccount}},
+		}}
+		output, err := a.ec2Client.DescribeImages(&input)
+		if err != nil {
+			return "", fmt.Errorf("failed to describe image with name %s and owner %s: %v", name, ownerAccount, err)
+		}
+		if len(output.Images) != 1 {
```

Comment:
When search for AMIs by name and owner, isn't it expected that we get multiple back from AWS (and then select the most recent one?)

Reply:
Yes in the normal case but I am assuming here that the image names are immutable because of the pipelines.

Reply:
So it's something like `kubernetes-image-ami-master-42`?

Reply:
Yes